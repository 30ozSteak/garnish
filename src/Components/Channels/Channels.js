import React from "react";

import "../../Components/Featured/Featured";

export const Channels = ({ channelName, channelString, handleNews, type }) => (
  <div onClick={() => handleNews(type)} className="channel-box">
    <div className="channel-button">
      <p>{channelName}</p>
      <p className="channel-desc">{channelString}</p>
    </div>
  </div>
);

export default Channels;
