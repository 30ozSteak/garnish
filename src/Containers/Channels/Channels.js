import React from "react";
import { connect } from "react-redux";
import { getChannel, activateChannel } from "../../Actions/index";
import "../../Components/Featured/Featured";

export const Channel = ({ channelName, channelString, onClick, active }) => (
  <div onClick={onClick} className="channel-box">
    <div className="channel-button">
      <p>{channelName}</p>
    </div>
  </div>
);

export const mapStateToProps = state => ({
  active: state.channel
});

export const mapDispatchToProps = (dispatch, ownProps) => ({
  onClick: () => {
    dispatch(getChannel(ownProps.channelString));
  }
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Channel);
