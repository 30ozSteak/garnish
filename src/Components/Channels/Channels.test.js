import React from "react";
import Channels from "./Channels";
import { shallow } from "enzyme";

describe("Channels", () => {
  it("should resemble the snapshot", () => {
    const wrapper = shallow(<Channels />);
    expect(wrapper).toMatchSnapshot();
  });
});
