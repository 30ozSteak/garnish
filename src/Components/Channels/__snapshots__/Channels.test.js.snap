// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Channels should resemble the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Channels />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="channel-button"
      >
        <p />
        <p
          className="channel-desc"
        />
      </div>,
      "className": "channel-box",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p />,
          <p
            className="channel-desc"
          />,
        ],
        "className": "channel-button",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "channel-desc",
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="channel-button"
        >
          <p />
          <p
            className="channel-desc"
          />
        </div>,
        "className": "channel-box",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p />,
            <p
              className="channel-desc"
            />,
          ],
          "className": "channel-button",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "channel-desc",
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
